datafusion/expr/src/logical_plan/dml.rs

```diff
@@ -35,8 +35,8 @@ pub struct CopyTo {
     pub output_url: String,
     /// Determines which, if any, columns should be used for hive-style partitioned writes
     pub partition_by: Vec<String>,
-    /// File format options.
-    pub format_options: FormatOptions,
+    /// File type trait
+    pub file_type: Arc<dyn FileType>,
```

Comment:
If using DataFusion `core` for execution plans, this `FileType` can be converted to a `FileFormatFactory` which will produce a `FileFormat` at runtime.

Reply:
What is the reason it can't be `FileFormatFactory` directly 🤔

Reply:
This would introduce a dependency between datafusion-expr and datafusion-core. The `FileType` trait exists to avoid this dependency and so users only using datafusion for logical planning would not have to implement all of the physical execution related `FileFormat`methods.